azext_edge/tests/edge/orchestration/resources/registry_endpoint/test_registry_endpoints_int.py

```diff
@@ -0,0 +1,592 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import pytest
+
+from azext_edge.tests.generators import generate_random_string
+from azext_edge.tests.helpers import run
+from azext_edge.tests.settings import EnvironmentVariables
+
+# pytest mark for rpsaas (cloud-side) tests
+pytestmark = pytest.mark.rpsaas
+
+
+@pytest.fixture(scope="function")
+def registry_endpoint_test_setup(settings):
+    """Setup fixture for registry endpoint tests."""
+    settings.add_to_config(EnvironmentVariables.rg.value)
+    settings.add_to_config(EnvironmentVariables.instance.value)
+    if not all([settings.env.azext_edge_instance, settings.env.azext_edge_rg]):
+        raise AssertionError(
+            "Cannot run registry endpoint tests without an instance and resource group. "
+            f"Current settings:\n {settings}"
+        )
+
+    yield {
+        "resourceGroup": settings.env.azext_edge_rg,
+        "instanceName": settings.env.azext_edge_instance,
+    }
+
+
+def test_registry_endpoint_lifecycle_anonymous(
+    registry_endpoint_test_setup, tracked_resources
+):
+    """Test complete lifecycle of registry endpoint with Anonymous authentication."""
+    resource_group = registry_endpoint_test_setup["resourceGroup"]
+    instance_name = registry_endpoint_test_setup["instanceName"]
+    registry_endpoint_name = (
+        f"test-registry-{generate_random_string(force_lower=True, size=8)}"
+    )
+    host = "myregistry.azurecr.io"
```

Comment:
Is the form "https://myregistry.azurecr.io" supposed to work? If not maybe change the description of the parameter from "The URL endpoint of the Azure Container Registry." -> "The endpoint of the Azure Container Registry." or "The hostname of the Azure Container Registry." when the user doesn't provide a scheme.